source/agora/consensus/data/Transaction.d

```diff
@@ -77,6 +88,8 @@ public struct Transaction
 
     public void serialize (scope SerializeDg dg) const @safe
     {
+        serializePart(this.type, dg);
```

Comment:
Deserialize it here directly using `OriginalType`

Reply:
Using OriginalType